tests/notification.spec.ts

```diff
@@ -1,40 +1,60 @@
-        const newIdea = await api(request).post("/api/idea-cards", {
-            ideaName: "Testing2",
-            tagline: "Yay!",
-            description: "I am testing",
-            targetAudience: "Yes",
-            features: "Yes",
-            experience: "Yes",
-            extraInfo: "Yes",
-            involveLevel: "minimum",
-            status: "workshopping",
+let user1Jwt, user2Jwt, notificationId;
+
+test.describe('/api/notification policy', () => {
+    test.beforeAll(async ({ request }) => {
+        const apiContext = api(request);
+
+        // Authenticate user1
+        const user1LoginResponse = await apiContext.post('/api/auth/local', {
+            identifier: config.user.username,
+            password: config.user.password,
+        });
+        user1Jwt = user1LoginResponse.jwt;
+
+        // Authenticate user2
+        const user2LoginResponse = await apiContext.post('/api/auth/local', {
+            identifier: config.user2.username,
+            password: config.user2.password,
+        });
+        user2Jwt = user2LoginResponse.jwt;
+
+        // Create a notification for user1
+        const notificationResponse = await apiContext.post('/api/notifications', {
+            createdDateTime: new Date().toISOString(),
+            readDateTime: null,
+            user: user1LoginResponse.user.id,
         });
-        ideaId = newIdea.id;
-        const notifications = await api(request).get("/api/notifications");
-        const notification = notifications.data
-        .find(item => (item.attributes.entityId === newIdea.id && item.attributes.entityType === "IdeaCard"));
-        expect(notification.attributes.Title).toBe("A new idea has been created");
-        expect(notification.attributes.Content).toBe("You created idea Testing2");
+        notificationId = notificationResponse.id;
     });
 
-    test.fixme("Create new notification on comment create", async ({ request }) => {
-        const newComment = await api(request).post("/api/comments", {
-            author: "tester2",
-            idea_card: { id: ideaId },
-            text: "test comment"
+    test("Ensure notifications are filtered by policy", async ({ request }) => {
+        const apiContext = api(request);
+
+        // Fetch notifications as user1
+        const user1NotificationsResponse = await request.get('/api/notifications', {
+            headers: {
+                Authorization: `Bearer ${user1Jwt}`,
+            },
         });
-        const notifications = await api(request).get("/api/notifications");
-        const notification = notifications.data
-        .find(item => (item.attributes.entityId === newComment.id && item.attributes.entityType === "Comment"));
-        expect(notification.attributes.Title).toBe("tester2 commented on Testing2");
-        expect(notification.attributes.Content).toBe("test comment");
+        const user1Notifications = await user1NotificationsResponse.json();
+
```

Comment:
Is it necessary to convert the response to json?

If yes, Why the api is returning another response type different of json?